### PR TITLE
Avoid crash on connecting to Jabber server

### PR DESCRIPTION
**Pull request by mcepl**
See https://bugzilla.redhat.com/show_bug.cgi?id=668190

Caused by our malfunctioning implementation of srv_lookup.

### CODE REVIEW

**Comment by dequis** [comment]
Uh, the linked bug has a patch from 2011 which i can't quite tell what it does because it changes the indentation of the file. Is that one related?

How do I reproduce this issue? Grab centos5 box, build bitlbee, install, connect to any jabber server?

File: lib/misc.c [hunk 1]
```diff
@@ -755,3 +842,24 @@ int truncate_utf8(char *string, int maxlen)
 	*end = '\0';
 	return end - string;
 }
+
+#ifdef TEST
+int main()
```
Comment (dequis) [inline lib/misc.c]: Test code doesn't belong here, there's a ./tests/ directory for that.

**Comment by dequis** [comment]
Also I'm not really sure how that bug survived after 632627e1ead19fc7dc07effe441b2d543a675229

Possibly related issue from my personal issue tracker: https://github.com/dequis/bitlbee/issues/58 (only affects debian lenny, which is almost as old as centos 5)

**Comment by dequis** [comment]
Oh uh, almost forgot, we dropped support for centos5 because of the glib version bump. Is this even relevant anymore?

**Comment by mcepl** [comment]
> How do I reproduce this issue? Grab centos5 box, build bitlbee, install, connect to any jabber server?

Yes, that should be more or less it.

**Comment by dequis** [comment]
Ehh, time to admit this one isn't going anywhere.

**Pull request closed without merging**